active/AnimatedIcon/AnimatedIconSpec.md

```diff
@@ -0,0 +1,282 @@
+
+# How to design animations for AnimatedIcon
+
+## Creating animated content with Lottie 
+
+1. Download the Lottie file for the icon you are looking to add to your application.
+2. Run the Lottie file through LottieGen. The LottieGen will output the file that you will add to your project that will include the markers needed for animated icon to map the correct state transitions to the correct animation segments. It will also contain the themable color properties you can set in code. 
+3. Take the output file from LottieGen and place it in your project.
+
+## Design Requirements  
+
+Designers will need to add markers named like the following to all Lottie files that will be used with AnimatedIcon.
+The marker names should reflect the visual state the animation is going from to the visual state the animation is
+going to. For example, the bare minimum of the markers that are needed for icons being used in a Button are:  
+
+* NormalToPointerOver 
+* NormalToPressed 
+* PointerOverToNormal 
+* PointerOverToPressed 
+* PressedToNormal 
+* PressedToPointerOver  
+
+“Normal” is the state where the user is not interacting with the button. 
+
+Markers will need to be named this specific way so AnimatedIcon can map the correct animation segment with the correct visual state transition. This solution is conforming to a standard set of strings that are already defined in the visual state manager for each state. Here is an example of how the markers would look in a Lottie JSON file for a button.  
+
+"markers":[{"tm":0,"cm":"NormalToPointerOver_Start","dr":0},{"tm":9,"cm":"NormalToPointerOver_End","dr":0}, 
+
+{"tm":10,"cm":"NormalToPressed_Start","dr":0},{"tm":19,"cm":"NormalToPressed_End","dr":0}, 
+
+{"tm":20,"cm":"PointerOverToNormal_Start","dr":0},{"tm":29,"cm":"PointerOverToNormal_End","dr":0}, 
+
+{"tm":30,"cm":"PointerOverToPressed_Start","dr":0},{"tm":39,"cm":"PointerOverToPressed_End","dr":0}, 
+
+{"tm":40,"cm":"PressedToNormal_Start","dr":0},{"tm":49,"cm":"PressedToNormal_End","dr":0}, 
+
+{"tm":50,"cm":"PressedToPointerOver_Start","dr":0},{"tm":69,"cm":"PressedToPointerOver_End","dr":0}, 
+
+{"tm":90,"cm":"PressedToNormal_Start","dr":0},{"tm":99,"cm":"PressedToNormal_End","dr":0}, 
+
+{"tm":100,"cm":"PressedToPointerOver_Start","dr":0},{"tm":101,"cm":"PressedToPointerOver_End","dr":0}] 
+
+
+
+# API Pages
+
+## AnimatedIcon class
+
+An AnimatedIcon is a UI component which plays animated images in response to user interaction and visual state changes.
+
+On Windows, the VisualStateManager is used to manage the logic for transitioning between states for controls.
+Each control has a specified VisualStateGroup, which contains mutually exclusive VisualState objects.
+For example, a Button may have a VisualStateGroup called “CommonStates”, which specifies a VisualState object for
+“Normal” “Pressed”, “PointerOver”, and “Disabled”.
+
+A control will use the VisualStateManager to determine what state the animated icon is transitioning to and
```

Comment:
What AnimatedIcon API does the control call to play the animation segment?

Reply:
You set the state property on AnimatedIcon and the property changed handler figures out what segement to play and plays it immediately (technically, it waits one layout cycle).

Reply:
Can we add some sample code? (Especially useful for the AnimatedIconSource case.)

Reply:
I included an example that is for a control that already has the default template updated and an example of adding an AnimatedIcon to a control that you would need to update the template to.